### PR TITLE
Consolidate rust_prost_library and fix extension-only proto generation.

### PR DESCRIPTION
**Pull request by freeformstu**
This PR consolidates rust_prost_library and rust_tonic_library into one rule; `rust_prost_library`. `rust_prost_library` will generate tonic services if you provide a `tonic_plugin` in the toolchain definition. If you do not provide that plugin and you try to build a proto with services, it will print a warning that you should add a `tonic_plugin`.

This PR also handles extension-only proto files. Prost does not generate a file if there are no messages, enums, or services and it appears that Prost doesn't even support proto2 extensions. So to work around this issue, protoc_wrapper will generate an empty `.rs` file in the case that there are only extensions defined in a file.

Closes: #2046

### CODE REVIEW

**Comment by freeformstu** [comment]
@UebelAndre & @illicitonion, please take a look when you have a chance.

**Comment by aaronmondal** [comment]
I'm amazed by the speed at which you come up with these fixes :heart_eyes: Thank you so much for your efforts! :heart:

UX-wise I think this simplified API is a good change.

**Review by illicitonion — APPROVED** [review]

**Comment by illicitonion** [comment]
Thanks!

File: proto/prost/private/protoc_wrapper.rs [hunk 1]
```diff
@@ -509,6 +501,11 @@ impl Args {
                             .expect("Failed to read file")
                             .lines()
                         {
+                            if crate_name.as_ref().unwrap() == "annotations_proto"
```
[the remark this answers is not in the document]
Comment (freeformstu) [reply proto/prost/private/protoc_wrapper.rs:504]: Ah shoot, that's my bad. This is left over from when I was debugging potential fixes. I'm pushing a PR to fix this now.

[the remark this answers is not in the document]
Comment (freeformstu) [reply proto/prost/private/protoc_wrapper.rs:504]: https://github.com/bazelbuild/rules_rust/pull/2048

**Pull request merged**